client/python/cryoet_data_portal/Makefile

```diff
@@ -1,6 +1,6 @@
 .PHONY: test-infra
 test-infra:
-	cd tests && docker compose up -d
+	cd tests && docker compose up -d --wait --wait-timeout 300
```

Comment:
Why add this here? You can add timeout information to the [docker-compose](https://docs.docker.com/reference/compose-file/services/#healthcheck):
```
  interval: 1m30s
  timeout: 10s
  retries: 3
  start_period: 40s
  start_interval: 5s
  ```

Reply:
I wasn't aware of this - thanks!

After a first read, those options seem to be about controlling how the healthcheck status is updated in an ongoing way. E.g. `interval` controls how frequently the test is run and `timeout` specifies how long the test can run for before being deemed unhealthy.

So I'm not sure how I could use those to ensure that `docker compose up` gives up waiting for a healthy status after a certain duration (i.e. I'm trying to protect against a runner indefinitely hanging due to this specific step). I understood that `--wait` will exit exactly once all services are healthy and will wait indefinitely if any service is not reported as healthy (unless you specify the `--wait-timeout` option too).

If you have a set of healthcheck option values that would be equivalent or give us what we want here, please share. Sorry if I'm being slow here!

Reply:
That makes perfect sense.